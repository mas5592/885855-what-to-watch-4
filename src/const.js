export const ENTRY_POINT = `https://4.react.pages.academy/wtw`;

export const AVATAR_URL = `https://4.react.pages.academy/wtw/asda.jpg`;

export const TIMEOUT_DURATION = 5000;

export const MAX_SIMILAR_CARDS = 4;

export const LIMIT_FILMS_COUNT = 8;

export const RATINGS_NUMBER = 5;

export const FILTER_ALL_GENRES = `All genres`;

export const PageNames = {
  MAIN: `main`,
  FILM_DETAILS: `Details`,
  SIGN_IN: `sign in`,
  ADD_REVIEW: `add review`,
  MY_LIST: `my list`,
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  PAGE: `/films`,
  FILM_REVIEW: `/review`,
  MY_LIST: `/mylist`,
  VIDEO_PLAYER: `/player`,
};

export const RatingLevel = {
  AWESOME: `Awesome`,
  VERY_GOOD: `Very good`,
  GOOD: `Good`,
  NORMAL: `Normal`,
  BAD: `Bad`
};

export const ListFilters = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const Time = {
  MINUTES_IN_HOUR: 60,
  SECONDS_IN_MINUTE: 60,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ReviewLength = {
  MIN: 50,
  MAX: 400,
};

export const reviewSubmitBtn = {
  post: `Post`,
  sending: `Sending...`,
};
