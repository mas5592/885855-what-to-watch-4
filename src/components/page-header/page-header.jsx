import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {AppRoute, AuthorizationStatus, PageNames} from '../../const';
import {getUserInfo, getAuthorizationStatus} from '../../reducer/user/selectors.js';

const PageHeader = ({currentPage, isSignedIn, userInfo, children}) => {

  const isSignInPage = currentPage === PageNames.SIGN_IN;
  const isMyListPage = currentPage === PageNames.MY_LIST;
  const isWithBreadcrumbs = currentPage === PageNames.ADD_REVIEW;

  const isWithTitle = isSignInPage || isMyListPage;

  const pageTitleElement = (
    <h1 className="page-title user-page__title">
      {isSignInPage && `Sign in`}
      {isMyListPage && `My list`}
    </h1>
  );

  const userBlockElement = (
    <div className="user-block">
      {isSignedIn &&
        <Link to={AppRoute.MY_LIST}>
          <div className="user-block__avatar">
            <img
              src={userInfo.avatarUrl}
              alt={userInfo.name}
              width={63}
              height={63}
            />
          </div>
        </Link>}
      {!isSignedIn &&
        <Link
          to={AppRoute.LOGIN}
          className="user-block__link"
        >
        Sign in
        </Link>}
    </div>
  );

  return (
    <header className={`page-header ${isWithTitle ? `user-page__head` : `movie-card__head`}`}>
      <div className="logo">
        <Link to={AppRoute.ROOT}
          className="logo__link"
        >
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>

      {isWithBreadcrumbs && children}
      {isSignInPage || isMyListPage ? pageTitleElement : null}
      {!isSignInPage && userBlockElement}
    </header>
  );
};

PageHeader.propTypes = {
  currentPage: PropTypes.string,
  isSignedIn: PropTypes.bool,
  userInfo: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
    name: PropTypes.string,
    avatarUrl: PropTypes.string,
  }),
  children: PropTypes.element,
};

const mapStateToProps = (state) => ({
  isSignedIn: getAuthorizationStatus(state) === AuthorizationStatus.AUTH,
  userInfo: getUserInfo(state),
});

export {PageHeader};
export default connect(mapStateToProps)(PageHeader);
