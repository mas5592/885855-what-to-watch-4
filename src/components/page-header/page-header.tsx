import * as React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {AppRoute, AuthorizationStatus, PageNames} from '../../const';
import {getUserInfo, getAuthorizationStatus} from '../../reducer/user/selectors';
import {FilmType, UserType} from '../../types';

interface Props {
  currentPage: FilmType;
  isSignedIn: boolean;
  userInfo: UserType;
  children?: JSX.Element;
}

const PageHeader: React.FC<Props> = (props: Props) => {
  const {
    currentPage,
    children,
    isSignedIn,
    userInfo,
  } = props;

  const isSignInPage = currentPage === PageNames.SIGN_IN;
  const isMyListPage = currentPage === PageNames.MY_LIST;
  const isWithBreadcrumbs = currentPage === PageNames.ADD_REVIEW;

  const isWithTitle = isSignInPage || isMyListPage;

  const pageTitleElement = (
    <h1 className="page-title user-page__title">
      {isSignInPage && `Sign in`}
      {isMyListPage && `My list`}
    </h1>
  );

  const userBlockElement = (
    <div className="user-block">
      {isSignedIn &&
        <Link to={AppRoute.MY_LIST}>
          <div className="user-block__avatar">
            <img
              src={userInfo.avatarURL}
              alt={userInfo.name}
              width={63}
              height={63}
            />
          </div>
        </Link>}
      {!isSignedIn &&
        <Link
          to={AppRoute.LOGIN}
          className="user-block__link"
        >
        Sign in
        </Link>}
    </div>
  );

  return (
    <header className={`page-header ${isWithTitle ? `user-page__head` : `movie-card__head`}`}>
      <div className="logo">
        <Link to={AppRoute.ROOT}
          className="logo__link"
        >
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>

      {isWithBreadcrumbs && children}
      {isSignInPage || isMyListPage ? pageTitleElement : null}
      {!isSignInPage && userBlockElement}
    </header>
  );
};

const mapStateToProps = (state) => ({
  isSignedIn: getAuthorizationStatus(state) === AuthorizationStatus.AUTH,
  userInfo: getUserInfo(state),
});

export {PageHeader};
export default connect(mapStateToProps)(PageHeader);
