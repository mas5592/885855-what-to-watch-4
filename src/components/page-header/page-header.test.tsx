import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import history from '../../history';
import {PageNames} from '../../const';
import {PageHeader} from './page-header';
import {AVATAR_URL} from '../../const.js';

const userInfo = {
  id: 1,
  email: `sadas@dsasd.ru`,
  name: `asdasd`,
  avatarURL: AVATAR_URL,
};

describe(`PageHeader`, () => {
  it(`Should be displayed correctly when the main page and the user are logged in`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <PageHeader
              currentPage={PageNames.MAIN}
              isSignedIn={true}
              userInfo={userInfo}
            />
          </Router>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should be displayed correctly when the main page is located and the user is not logged in`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <PageHeader
              currentPage={PageNames.ADD_REVIEW}
              isSignedIn={false}
              userInfo={userInfo}
            />
          </Router>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should be displayed correctly when it is not the main page or the login page`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <PageHeader
              currentPage={PageNames.ADD_REVIEW}
              isSignedIn={true}
              userInfo={userInfo}
            />
          </Router>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should be displayed correctly when you log in to the page`, () => {
    const tree = renderer
      .create(
          <Router history={history}>
            <PageHeader
              currentPage={PageNames.SIGN_IN}
              isSignedIn={false}
              userInfo={userInfo}
            />
          </Router>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
