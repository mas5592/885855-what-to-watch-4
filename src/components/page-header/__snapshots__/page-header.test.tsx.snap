// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Should be displayed correctly when it is not the main page or the login page 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="asdasd"
          height={63}
          src="https://4.react.pages.academy/wtw/asda.jpg"
          width={63}
        />
      </div>
    </a>
  </div>
</header>
`;

exports[`PageHeader Should be displayed correctly when the main page and the user are logged in 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="asdasd"
          height={63}
          src="https://4.react.pages.academy/wtw/asda.jpg"
          width={63}
        />
      </div>
    </a>
  </div>
</header>
`;

exports[`PageHeader Should be displayed correctly when the main page is located and the user is not logged in 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign in
    </a>
  </div>
</header>
`;

exports[`PageHeader Should be displayed correctly when you log in to the page 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    Sign in
  </h1>
</header>
`;
